pkg/azuredx/client/client.go

```diff
@@ -237,15 +238,15 @@ func (c *Client) KustoRequest(ctx context.Context, clusterUrl string, path strin
 	switch {
 	case resp.StatusCode == http.StatusUnauthorized:
 		// HTTP 401 has no error body
-		return nil, fmt.Errorf("azure HTTP %q", resp.Status)
+		return nil, errorsource.DownstreamError(fmt.Errorf("azure HTTP %q", resp.Status), false)
 
 	case resp.StatusCode/100 != 2:
 		var r models.ErrorResponse
 		err := json.NewDecoder(resp.Body).Decode(&r)
 		if err != nil {
 			return nil, fmt.Errorf("azure HTTP %q with malformed error response: %s", resp.Status, err)
 		}
-		return nil, fmt.Errorf("azure HTTP %q: %q.\nReceived %q: %q", resp.Status, r.Error.Message, r.Error.Type, r.Error.Description)
+		return nil, errorsource.DownstreamError(fmt.Errorf("azure HTTP %q: %q.\nReceived %q: %q", resp.Status, r.Error.Message, r.Error.Type, r.Error.Description), false)
```

Comment:
If status code is not 2XX, we usually use [ErrorSourceFromHTTPStatus](https://github.com/grafana/grafana-plugin-sdk-go/blob/b8e7e713c62697cc58df8572a9ebc1880cfa4733/backend/error_source.go#L14) to determine error source.